apiserver/common.go

```diff
@@ -125,6 +122,42 @@ func IsValidPVC(pvcName, ns string) bool {
 	return pvc != nil
 }
 
+// ValidateResourceRequestLimit validates that a Kubernetes Requests/Limit pair
+// is valid, both by validating the values are valid quantity values, and then
+// by checking that the limit >= request. This also needs to check against the
+// configured values for a request, which must be provided as a value
+func ValidateResourceRequestLimit(request, limit string, defaultQuantity resource.Quantity) error {
+	// ensure that the request/limit are valid, as this simplifies the rest of
+	// this code. We know that the defaultRequest is already valid at this point,
+	// as otherwise the Operator will fail to load
+	if err := ValidateQuantity(request); err != nil {
+		return err
+	}
+
+	if err := ValidateQuantity(limit); err != nil {
+		return err
+	}
+
+	// parse the quantities so we can compare
+	requestQuantity, _ := resource.ParseQuantity(request)
+	limitQuantity, _ := resource.ParseQuantity(limit)
+
+	// if limit and request are nonzero and the limit is less than the request,
+	// error
+	if !limitQuantity.IsZero() && !requestQuantity.IsZero() && limitQuantity.Cmp(requestQuantity) == -1 {
+		return fmt.Errorf(ErrMessageLimitInvalid, limitQuantity.String(), requestQuantity.String())
+	}
+
+	// if limit and the default are nonzero, but the request is zero, and the
+	// limit is less than the default, return an error
+	if !limitQuantity.IsZero() && requestQuantity.IsZero() && !defaultQuantity.IsZero() &&
+		limitQuantity.Cmp(defaultQuantity) == -1 {
+		return fmt.Errorf(ErrMessageLimitInvalid, limitQuantity.String(), defaultQuantity.String())
+	}
```

Comment:
This condition is a lot to take in and the comment doesn't explain why it's here. Just from reading, I think these lines can be eliminated by setting request to the default before the check on L147:

```go
	// parse the quantities so we can compare
	requestQuantity, _ := resource.ParseQuantity(request)
	limitQuantity, _ := resource.ParseQuantity(limit)

	if requestQuantity.IsZero() {
		requestQuantity = defaultQuantity
	}

	// if limit and request are nonzero and the limit is less than the request,
	// error
	if !limitQuantity.IsZero() && !requestQuantity.IsZero() && limitQuantity.Cmp(requestQuantity) == -1 {
		return fmt.Errorf(ErrMessageLimitInvalid, limitQuantity.String(), requestQuantity.String())
	}
```

1. My reading could be wrong. Would the existing tests pass with this change? Is this the intended behavior?
2. If so, using the default this way makes sense to me and seems more clear. Is there another reason to keep the current arrangement?